src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -239,16 +239,16 @@ public HostOutputMessage HostOutputMessage
                         traceWriter,
                         instance.FunctionDescriptor);
                     var valueBindingContext = new ValueBindingContext(functionContext, cancellationToken);
-                    var parameters = await instance.BindingSource.BindAsync(valueBindingContext);
-
+                    await parameterHelper.BindAsync(instance.BindingSource, valueBindingContext);
+                    
                     Exception invocationException = null;
                     ExceptionDispatchInfo exceptionInfo = null;
                     string startedMessageId = null;
-                    using (ValueProviderDisposable.Create(parameters))
+                    using (parameterHelper.GetDisposables())
```

Comment:
GetDisposables is a little strange. Can't the entire ParameterHelper be IDisposable, so you can just do `using(parameterHelper)` here?